lib/events.js

```diff
@@ -306,7 +306,8 @@ EventEmitter.prototype.removeListener =
         else {
           delete events[type];
           if (events.removeListener)
-            this.emit('removeListener', type, listener);
+            this.emit('removeListener', type,
```

Comment:
@indutny Thanks for having a look. That would probably do fine for the "there is a single listener right now" case, but I'm not sure it would work out for when `list` is actually a list, given:

* passed in `listener` could be either the wrapped listener, or the unwrapped one (internal call vs user call)
* the unwrapped listener _could_ be used multiple times, either for `once` and `on` or multiple `once`s or whatever, and so we should prefer to match the wrapped fn

So we could remove repetition by doing `const listenerToEmit = listener.listener ? listener.listener : listener;` at the top, but you aren't necessarily (or as I understand it, likely) going to even need that (not the common case to have `removeListener` events).